rooms/room_main/room-main.gd

```diff
@@ -28,22 +49,47 @@ func _ready() -> void:
 	#	Quit
 	nUIMainMenu.menu_quit.connect( Callable( nSignals, "_on_menu_quit" ) )
 	nUIMainMenu.menu_focus()
+	print( nUIMainMenu )
+
+
+func _ready() -> void:
+	#	Need to know if we've configured settings before.
+	#	Config files will be located in:
+	#	%AppData%\Roaming\Godot\app_userdata\GDTemplate
+	if( loaded_settings == false ):
+		GlobalUserSettings.set_first_time_setup( false )
+	#	If we couldn't find the loaded files or if you saw the menu already:
+	if( GlobalUserSettings.first_time_setup == false ):
+		nUIFirstSetup = pUIFirstSetup.instantiate()
+		nUI.add_child( nUIFirstSetup )
+		nUIFirstSetup.new_language.connect(
+				Callable( nSignals, "_on_new_language" ) )
+		nUIFirstSetup.completed_first_setup.connect(
+				Callable( nSignals, "_on_completed_first_setup" ) )
+	else:
+		add_main_menus()
 	#	Mainly have to do this just so that the language adjusts correctly.
 	TranslationServer.set_locale(
 			GlobalUserSettings.accessibility[ "current_language" ] )
 
 
 func destroy() -> void:
 	#	Should run all destroy functions that can be located in game.
-	nUIMainMenu.menu_new_game.disconnect(
-			Callable( nSignals, "_on_menu_new_game" ) )
-	nUISettings.new_language.disconnect(
-			Callable( nSignals, "_on_new_language" ) )
-	nUISettings.menu_settings_closed.disconnect(
-			Callable( nSignals, "_on_menu_settings_closed" ) )
-	nUIMainMenu.menu_settings.disconnect( 
-			Callable( nSignals, "_on_menu_settings" ) )
-	nUIMainMenu.menu_quit.disconnect( Callable( nSignals, "_on_menu_quit" ) )
-	nUISettings.destroy()
+	if( nUIFirstSetup != null ):
+		nUIFirstSetup.disconnect_first_setup_signals()
```

Comment:
Is this function actually getting called??

Reply:
It is not. Forgot to test this case. Corrected to refer to nSignals, where the signals get disconnected. For consistency I'll apply this change to the other menus, as well.